### PR TITLE
Modify S5807: LaYC format

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5807/python/rule.adoc [hunk 1]
```diff
@@ -1,19 +1,24 @@
+This rule raises an issue when a name listed in the ``++__all__++`` property of a module has not been defined.
+
 == Why is this an issue?
 
-Developers may define a list named ``++__all__++`` in a module to limit the names imported from it by wildcard imports (``++from mymodule import *++``). This list can only reference defined names, otherwise an ``++AttributeError++`` will be raised when the module is imported.
+The ``++__all__++`` property of a module is used to define the list of names that will be imported when that module is imported through a wildcard import, i.e. when ``++from mymodule import *++`` is used.
+
```
Comment (joke1196) [inline rules/S5807/python/rule.adoc:6]: I find this sentence a bit confusing, especially with the repetition of `imported`. Maybe we could just add a small example to illustrate it better. Something like `fileX.py` as the `from moduleY import *` statement and in `moduleY/__init__.py` we would have the ``++__all__++`` list. What do you think?

**Review by joke1196 — CHANGES_REQUESTED** [review]
LGTM! I have just added a small comment about a potential example we could add. Tell me what you think.

File: rules/S5807/python/rule.adoc (hunk 1 above)
Comment (guillaume-dequenne-sonarsource) [reply rules/S5807/python/rule.adoc]: I added an example, please let me know what you think.

**Review by joke1196 — APPROVED** [review]
LGTM. I think the example is quite nice!

**Review by jean-jimbo-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**